src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -90,13 +90,19 @@ protected async Task<IEnumerable<AvailableCodeAction>> GetAvailableCodeActions(I
 
             var distinctActions = codeActions.GroupBy(x => x.Title).Select(x => x.First());
 
-            // Be sure to filter out any code actions that inherit from CodeActionWithOptions.
-            // This isn't a great solution and might need changing later, but every Roslyn code action
-            // derived from this type tries to display a dialog. For now, this is a reasonable solution.
-            var availableActions = ConvertToAvailableCodeAction(distinctActions)
-                .Where(a => !a.CodeAction.GetType().GetTypeInfo().IsSubclassOf(typeof(CodeActionWithOptions)));
+            var availableActions = ConvertToAvailableCodeAction(distinctActions);
 
-            return availableActions;
+            return FilterBlacklistedCodeActions(availableActions);
+        }
+
+        private static IEnumerable<AvailableCodeAction> FilterBlacklistedCodeActions(IEnumerable<AvailableCodeAction> codeActions)
+        {
+            // Most of actions with UI works fine with defaults, however there's few exceptions:
+            return codeActions.Where(x =>
+                x.CodeAction.Title != "Generate new type..." &&         // Blacklisted because doesn't give additional value over non UI generate type (when defaults used.)
```

Comment:
let's do the blacklisting based on the original Roslyn type name instead of the title text